src/main/java/com/dawn/jat/illuminati/post/service/PostService.java

```diff
@@ -28,44 +27,73 @@
     @Autowired
     private PostSummaryRepository postSummaryRepository;
 
-    public List<PostSummaryEntity> findPostSummary() {
+    public List<PostSummaryEntity> getPostList() {
         return postSummaryRepository.findAll();
     }
 
-    public Optional<PostEntity> findBySlug(String slug) {
-        return postRepository.findBySlug(slug);
+    /**
+     * Get Post by Slug.
+     *
+     * @param slug String
+     * @return PostDto
+     */
+    public PostDto getPost(String slug) {
+        ModelMapper modelMapper = new ModelMapper();
+        return modelMapper.map(postRepository.findBySlug(slug), PostDto.class);
     }
 
     /**
      * Create new Entity from postDTO receive from client.
+     *
      * @param postDto PostDto
-     * @return PostEntity
+     * @return PostDto
      */
-    public PostEntity create(PostDto postDto) {
+    public PostDto createPost(PostDto postDto) {
         PostEntity post = new PostEntity();
         post = converter.convertPostDtoToEntity(postDto, post);
-        return postRepository.savePost(post);
+        postRepository.savePost(post);
+        postDto.setId(post.getId());
```

Comment:
if `savePost` return null or some thing happens we always return the received PostDto, we should  do something with the result from `postRepository.savePost(post);`